### ISSUE TITLE
Episodes and Podcasts menu issues

### ISSUE THREAD

**Issue by auouymous** [gpodder/gpodder#988]
The `play` item in episode context menu can be play, open, stream or preview. The Episodes menu has separate items for play and open, but lacks stream and preview. We should be able to remove the `Open` item and register a submenu-action for menuChannels that changes the `Play` item name using the code from context menu.

Download and Cancel are similar, they share the same item in context menu but have two items in the Episodes menu. Could use the menuChannels action like play.

Items added by extensions only appear in the context menu, not the Episodes menu. Could use the menuChannels action to add them.

Context menu has `New` and `Archive` with checkboxes, while the Episodes menu has `Toggle new status` and `Change delete lock`. These should be made consistent.

Context menu has `Send to`, the Episodes menu does not.

The Podcasts menu and channel context menu have similar issues.

Could these menus share the same code with their context menus to not only solve the above problems but prevent more in the future?

**Comment by elelay** [comment]
That's a good point. Global menus should offer the same options as context menus.

Regarding extensions, you would like an Extensions section in the Channels and Episodes menu? Seems OK to display the same as `on_episodes_context_menu` and `on_channel_context_menu`. It would have to be called on each selection change instead of context-menu trigger but they are not usually massive computations.

**Comment by auouymous** [comment]
> Regarding extensions, you would like an Extensions section in the Channels and Episodes menu?

Yes, so both menus are identical, because some users might not know about context menus or might find alt-e easier to press than shift-F10. And for me, the menu is scrolled when shift-F10 is used on a selection at bottom of window when window is at bottom of screen.

> It would have to be called on each selection change instead of context-menu trigger but they are not usually massive computations.

Why? A submenu-action is called when the Episodes menu is opened and closed, I just need to figure out how to detect between the opening and closing. That way calculations only happen when needed, instead of for every change of the selection, which most of the time won't have a menu opened.
